test/junit/scala/math/BigDecimalTest.scala

```diff
@@ -228,4 +228,35 @@ class BigDecimalTest {
   def test_SI8970() {
     assert((0.1).## == BigDecimal(0.1).##)
   }
+
+  // Motivated by the problem of MathContext lost
+  @Test
+  def testMathContext() {
+    def testPrecision() {
+      val e = 1000
+
+      val n = BigDecimal("1.1", MC.UNLIMITED).pow(e)
+      assert(BigDecimal(1.1d, MC.UNLIMITED).pow(e) == n)
+      assert(BigDecimal.decimal(1.1d, MC.UNLIMITED).pow(e) == n)
+      assert(BigDecimal.decimal(1.1f, MC.UNLIMITED).pow(e) == n)
+      assert(BigDecimal.decimal(new BD("1.1"), MC.UNLIMITED).pow(e) == n)
+
+      val m = BigDecimal(java.lang.Double.toString(1.1f.toDouble), MC.UNLIMITED).pow(e)
+      assert(BigDecimal(1.1f, MC.UNLIMITED).pow(e) == m) // deprecated
+
+      val l = BigDecimal("11", MC.UNLIMITED).pow(e)
```

Comment:
It might be cleaner to drop the creation of `l` and just test `(BigDecimal(11, MC.UNLIMITED)/10).pow`.